### PR TITLE
[libxml2] Update to v2.13.5

### PR DESCRIPTION
**Pull request by donny-dont**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [X] The "supports" clause reflects platforms that may be fixed by this new version.
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jimwang118** [comment]
Update PR [39478](https://github.com/microsoft/vcpkg/pull/39478) already exists.

**Comment by donny-dont** [comment]
> Update PR [39478](https://github.com/microsoft/vcpkg/pull/39478) already exists.

👋 @jimwang118  I’ve been trying to replace our custom registry at https://github.com/WebKitForWindows/WebKitRequirements with the canonical registry. Then I plan on integrating vcpkg into the build of WebKit.

I saw your PR but it seemed like you were having trouble with it. I have experience with building libxml2 and libxslt so that’s why I landed an update of the later in https://github.com/microsoft/vcpkg/commit/21692b2960b36564d42df3c3351ae7ddf211fb7b .

Comparing our two PRs I can see that it’s carrying over some stuff that’s obsolete. Like the pkgconfig specific changes. You can see they added stuff at https://github.com/GNOME/libxml2/blob/fe1ee0f25f43e33a9981fd6fe7b0483a8c8b5e8d/CMakeLists.txt#L385 which does the _LIBS changes. There’s also some commits added since the last release that obsolete some changes in that diff see https://github.com/GNOME/libxml2/commits/2.13/ .

There are also a number of build options that aren’t exposed in the feature listing that I was planning on adding to the vcpkg.json.

Apologies for not doing a thorough review of your PR making the above points. It’s after work hours for me and it’s hard to do a review through a phone.

**Comment by BillyONeal** [comment]
I reran the build to workaround a company-wide authentication issue that broke the first run but the failures now look legitimate:

```console
/Users/vcpkg/Data/b/librsvg/src/librsvg-2-a2e28f96fe.clean/rsvg-css.c:866:20: error: incompatible function pointer types assigning to 'xmlStructuredErrorFunc' (aka 'void (*)(void *, const struct _xmlError *)') from 'void (void *, xmlErrorPtr)' (aka 'void (void *, struct _xmlError *)') [-Wincompatible-function-pointer-types]
    handler.serror = rsvg_xml_noerror;
                   ^ ~~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

It seems likely that `rsvg_xml_noerror` needs to be patched to have a different signature due to changes in `libxml2`?

**Comment by donny-dont** [comment]
> I reran the build to workaround a company-wide authentication issue that broke the first run but the failures now look legitimate:
> 
> ```
> /Users/vcpkg/Data/b/librsvg/src/librsvg-2-a2e28f96fe.clean/rsvg-css.c:866:20: error: incompatible function pointer types assigning to 'xmlStructuredErrorFunc' (aka 'void (*)(void *, const struct _xmlError *)') from 'void (void *, xmlErrorPtr)' (aka 'void (void *, struct _xmlError *)') [-Wincompatible-function-pointer-types]
>     handler.serror = rsvg_xml_noerror;
>                    ^ ~~~~~~~~~~~~~~~~
> 1 warning and 1 error generated.
> ```
> 
> It seems likely that `rsvg_xml_noerror` needs to be patched to have a different signature due to changes in `libxml2`?

Yea I think the changes to that port from https://github.com/microsoft/vcpkg/pull/39478 need to be integrated. Will give that a go and add @jimwang118 as a co-author.

**Comment by donny-dont** [comment]
Alrighty added and credited @jimwang118 for his work. 🤞 were 🟢 and this can land

File: ports/libxml2/portfile.cmake [hunk 1]
```diff
@@ -21,6 +34,10 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
         "tools" LIBXML2_WITH_PROGRAMS
         "icu"  LIBXML2_WITH_ICU
 )
+
+vcpkg_find_acquire_program(PKGCONFIG)
+set(ENV{PKG_CONFIG} "${PKGCONFIG}")
+
```
Comment (jimwang118) [inline ports/libxml2/portfile.cmake:40]:
```suggestion

vcpkg_find_acquire_program(PKGCONFIG)
```
It is recommended to add option "-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}" in vcpkg_cmake_configure to pass in pkgconfig.

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: ports/libxml2/portfile.cmake (hunk 1 above)
Comment (donny-dont) [reply ports/libxml2/portfile.cmake]: Fixed!

**Review by jimwang118 — APPROVED** [review]

**Comment by LilyWangLL** [comment]
All features passed with following triplets:
```
x86-windows
x64-windows
x64-windows-static
```
Usage test passed on `x64-windows`.

**Review by LilyWangLL — APPROVED** [review]

**Comment by nirvn** [comment]
@donny-dont , I think this might have led to build failure on iOS, with the following error:

[3/46] /Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DLIBXML_STATIC -DSYSCONFDIR=\"/Users/runner/builddir/_deps/vcpkg-src/packages/libxml2_arm64-ios/etc\" -I/Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/arm64-ios-rel -I/Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/src/v2.13.5-a978bfa150.clean/include -isystem /Users/runner/builddir/vcpkg_installed/arm64-ios/include -fPIC -fvisibility=hidden -g -pedantic -Wall -Wextra -Wshadow -Wpointer-arith -Wcast-align -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wno-long-long -Wno-format-extra-args -O3 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode_15.4.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.5.sdk -miphoneos-version-min=14.0 -fPIC -MD -MT CMakeFiles/LibXml2.dir/dict.c.o -MF CMakeFiles/LibXml2.dir/dict.c.o.d -o CMakeFiles/LibXml2.dir/dict.c.o -c /Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/src/v2.13.5-a978bfa150.clean/dict.c
FAILED: CMakeFiles/LibXml2.dir/dict.c.o 
/Applications/Xcode_15.4.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DLIBXML_STATIC -DSYSCONFDIR=\"/Users/runner/builddir/_deps/vcpkg-src/packages/libxml2_arm64-ios/etc\" -I/Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/arm64-ios-rel -I/Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/src/v2.13.5-a978bfa150.clean/include -isystem /Users/runner/builddir/vcpkg_installed/arm64-ios/include -fPIC -fvisibility=hidden -g -pedantic -Wall -Wextra -Wshadow -Wpointer-arith -Wcast-align -Wwrite-strings -Wstrict-prototypes -Wmissing-prototypes -Wno-long-long -Wno-format-extra-args -O3 -DNDEBUG -arch arm64 -isysroot /Applications/Xcode_15.4.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.5.sdk -miphoneos-version-min=14.0 -fPIC -MD -MT CMakeFiles/LibXml2.dir/dict.c.o -MF CMakeFiles/LibXml2.dir/dict.c.o.d -o CMakeFiles/LibXml2.dir/dict.c.o -c /Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/src/v2.13.5-a978bfa150.clean/dict.c
/Users/runner/builddir/_deps/vcpkg-src/buildtrees/libxml2/src/v2.13.5-a978bfa150.clean/dict.c:970:17: error: call to undeclared function 'getentropy'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
            if (getentropy(globalRngState, sizeof(globalRngState)) == 0)
                ^
1 error generated.

**Comment by dg0yt** [comment]
IMO this is an upstream issue then. IIUC iOS doesn't have `getentropy`. The offending call is guarded with `HAVE_DECL_GETENTROPY`. So something goes wrong at configuration/detection of `HAVE_DECL_GETENTROPY`.

**Comment by nirvn** [comment]
@dg0yt , most likely yes, see this commit (https://gitlab.gnome.org/GNOME/libxml2/-/commit/e1657f3f272e87577aaf0c77aa040609394a0312).

The commit itself doesn't apply against 2.13.5; I've manually backported the commit into a patch locally and if that is successful I'll upstream.

**Pull request merged**